tests/sentry/workflow_engine/processors/test_delayed_workflow.py

```diff
@@ -356,3 +363,106 @@ def test_get_condition_query_groups(self):
         assert result[count_query] == {self.group.id, group2.id}  # count and percent condition
         assert result[percent_only_query] == {self.group.id}
         assert result[different_query] == {group3.id}
+
+
+@freeze_time(FROZEN_TIME)
+class TestGetSnubaResults(BaseWorkflowTest):
+    def tearDown(self):
+        super().tearDown()
+
+    def create_events(self, comparison_type: ComparisonType) -> Event:
+        # Create current events for the first query
+        event = self.create_event(self.project.id, FROZEN_TIME, "group-1", self.environment.name)
+        self.create_event(self.project.id, FROZEN_TIME, "group-1", self.environment.name)
+        if comparison_type == ComparisonType.PERCENT:
+            # Create a past event for the second query
+            self.create_event(
+                self.project.id,
+                FROZEN_TIME - timedelta(hours=1, minutes=10),
+                "group-1",
+                self.environment.name,
+            )
+        return event
+
+    def create_condition_groups(
+        self, data_conditions: list[DataCondition]
+    ) -> tuple[dict[UniqueConditionQuery, set[int]], int, list[UniqueConditionQuery]]:
+        condition_groups = {}
+        all_unique_queries = []
+        for data_condition in data_conditions:
+            unique_queries = generate_unique_queries(data_condition, self.environment.id)
+            comparison_type = (
+                ComparisonType.PERCENT
+                if Condition(data_condition.type) in PERCENT_CONDITIONS
+                else ComparisonType.COUNT
+            )
+            event = self.create_events(comparison_type)
+            assert event.group
+            group = event.group
+            condition_groups.update({query: {event.group.id} for query in unique_queries})
+            all_unique_queries.extend(unique_queries)
+        return condition_groups, group.id, all_unique_queries
+
+    def create_event_frequency_condition(
+        self, type: Condition | None = Condition.EVENT_FREQUENCY_COUNT
+    ) -> DataCondition:
+        if type not in SLOW_CONDITIONS:
+            raise ValueError(f"{type} is not a slow condition")
+
+        comparison = {"interval": "1h", "value": 100}
+
+        if type in PERCENT_CONDITIONS:
+            comparison["comparison_interval"] = "15m"
+
+        return self.create_data_condition(
+            type=type,
+            comparison=comparison,
+            condition_result=True,
+        )
+
+    def test_empty_condition_groups(self):
+        assert get_condition_group_results({}) == {}
+
+    def test_count_comparison_condition(self):
+        dc = self.create_event_frequency_condition()
+        condition_groups, group_id, unique_queries = self.create_condition_groups([dc])
+
+        results = get_condition_group_results(condition_groups)
+        assert results == {
+            unique_queries[0]: {group_id: 2},
```

Comment:
I forget exactly what this is - is it group id to number of events that match the query?

Reply:
it's key = unique condition query, value = result from snuba for a list of groups like `{group_id: count, group_id_2: count, etc}`

Reply:
where the groups in the result share the same query